### ISSUE TITLE
getComponentPropsFromTypeDefine error,when using interface to define a function

### ISSUE THREAD

**Issue by czb3279338858** [vuejs/eslint-plugin-vue#2120]
**Checklist**

- [*] I have tried restarting my IDE and the issue persists.
- [*] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** ^7.32.0
- **eslint-plugin-vue version:** ^8.0.3
- **Node version:** 16.14.2 64
- **Operating System:** window 10

**Please show your full configuration:**
```json5
module.exports = {
  root: true,
  env: {
    node: true
  },
  'extends': [
    'plugin:vue/essential',
    'eslint:recommended',
    '@vue/typescript/recommended'
  ],
  parserOptions: {
    ecmaVersion: 2020
  },
  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'warn' : 'off'
  }
}
```

**What did you do?**	
```vue
interface ABC {
  (a: string): string;
}
interface Props {
  propC: ABC;
}
withDefaults(defineProps<Props>(), {
  propC: (a) => "1",  // eslint error: Type of the default value for 'propC' prop must be a object
});
```

**What did you expect to happen?**
no error

**What actually happened?**
1、Type of the default value for 'propC' prop must be a object

2、The function getComponentPropsFromTypeDefine treats interface ABC as an object when processing it

3、I hope to export inferRuntimeType so that I can use it externally

**Repository to reproduce this issue**
The issue can be reproduced with this topic. What did you do?
